src/functionHandlers/notarisePdt/v2/handler.ts

```diff
@@ -40,21 +41,13 @@ export const main: Handler = async (
     fhirHelper.hasRequiredFields(data.type, parsedFhirBundle);
     fhirHelper.hasRecognisedFields(data.type, parsedFhirBundle);
   } catch (e) {
-    errorWithRef(
-      `Error while validating certificate: ${
-        e instanceof DetailedCodedError ? `${e.title}, ${e.messageBody}` : e
-      }`
+    const codedError = new CodedError(
+      "INVALID_CERTIFICATE",
+      "Error while validating certificate",
+      JSON.stringify(serializeError(e))
```

Comment:
This creation of a new CodedError() may not be necessary as `fhirHelper.hasRequiredFields()` and `fhirHelper.hasRecognisedFields()` are already going to be throwing CodedErrors.

Furthermore, when you `serializeError(e)`, you may be nesting CodedErrors 🤔

Shall we stick to creating them at just one place?

Reply:
Suggestion: Create an overall encompassing try/catch that does `e instanceof CodedError` checks before creating a `new CodedError()`. Then can remove the ones like these.

Reply:
For this file I will add an overall `try/catch` to wrap around the entire function. But I think I won't remove the other `try/catch` within the function because each of it represents the different stages (Validation, Endorsement, Sending to SPM, Google GPAY) of the function and it also shows different error types and messages for each stage initially.

**How it will look like will be something like:**
Within the inner try catch, I will throw the appropriate error to the most outer try catch
```
// try/catch within each stage
try{
...
}catch (e) {
        throw e instanceof CodedError
          ? e
          : new CodedError(
              "ERROR_TYPE",
              "ERROR_MESSAGE",
              JSON.stringify(serializeError(e))
            );
      }
```
 Then for the outer try catch I will return a response accordingly,
```
// overall encompassing try/catch
 try{
 // consisting of many inner try/catch for each stage
 ...
 } catch (e) {
    let codedError: CodedError;
    if (e instanceof CodedError) {
      codedError = e;
    } else {
      codedError = new CodedError(
        "UNKNOWN_ERROR",
        "Unable to notarise document",
        JSON.stringify(serializeError(e))
      );
    }

    errorWithRef(codedError.toString());
    return codedError.toResponse(reference);
  }
```